src/Uno.UI/UI/Xaml/Controls/AutoSuggestBox/AutoSuggestBox.cs

```diff
@@ -329,7 +329,16 @@ private void SubmitSearch()
 	    {
 			var finalResult = _suggestionsList.SelectedItem ?? GetObjectText(Text);
 
-			QuerySubmitted?.Invoke(this, new AutoSuggestBoxQuerySubmittedEventArgs(finalResult is "" ? null : finalResult, userInput));			
+			var isSelected = _suggestionsList.Items.FirstOrDefault(it => it?.ToString() == Text) is not null;
+
```

Comment:
This is assuming that the _suggestionsList will always be a `ListView` of `string`. 

Following the logic from the docs, we should never be setting the `ChosenSuggestion` property of the event args if we are firing this event as a result of the `_queryButton.Click` event.

https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.controls.autosuggestbox?view=winrt-22621#query-submitted

We should have something like `SubmitSearch(object chosenSuggestion = null)` instead. At the point of `SubmitSearch` being called, we shouldn't care whether the `userInput` matches a suggestion. The `userInput` is placed in the args regardless and the `AutoSuggestBoxQuerySubmittedEventArgs.ChosenSuggestion` will just be set to the optionally passed in `chosenSuggestion` .